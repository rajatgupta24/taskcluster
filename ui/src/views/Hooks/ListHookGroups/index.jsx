import React, { Component } from 'react';
import { graphql, withApollo } from 'react-apollo';
import { withStyles } from '@material-ui/core/styles';
import PlusIcon from 'mdi-react/PlusIcon';
import { parse, stringify } from 'qs';
import Typography from '@material-ui/core/Typography';
import Spinner from '../../../components/Spinner';
import Dashboard from '../../../components/Dashboard';
import HelpView from '../../../components/HelpView';
import Search from '../../../components/Search';
import Button from '../../../components/Button';
import ErrorPanel from '../../../components/ErrorPanel';
import HookGroupsTable from '../../../components/HookGroupsTable';
import hookGroupsQuery from './hookGroups.graphql';

@withApollo
@graphql(hookGroupsQuery, {
  options: {
    fetchPolicy: 'network-only',
  },
})
@withStyles(theme => ({
  actionButton: {
    ...theme.mixins.fab,
  },
  listItemProps: {
    button: true,
    color: '#fff',
  },
  hasErrors: {
    '& td, & a': {
      color: theme.palette.error.main,
    },
  },
  noFires: {
    '& td, & a': {
      color: theme.palette.warning.light,
    },
  },
}))
export default class ListHookGroups extends Component {
  handleCreateHook = () => {
    this.props.history.push('/hooks/create');
  };

  handleHookSearchSubmit = hookSearch => {
    const query = parse(window.location.search.slice(1));

    this.props.history.push({
      search: stringify({
        ...query,
        search: hookSearch,
      }),
    });
  };

  render() {
    const {
      classes,
      description,
      data: { loading, error, hookGroups },
    } = this.props;
    const { search } = parse(window.location.search.slice(1));
    const hookGroupIds = hookGroups?.map(group => group?.hookGroupId).flat();

    return (
      <Dashboard
        title="Hooks Groups"
        helpView={<HelpView description={description} />}
        search={
          <Search
            placeholder="Hook group contains"
            defaultValue={search}
            onSubmit={this.handleHookSearchSubmit}
          />
        }>
        {!hookGroups && loading && <Spinner loading />}
        <ErrorPanel fixed error={error} />
        {!loading &&
          (hookGroupIds?.length ? (
            <HookGroupsTable
              searchTerm={search}
              hookGroups={hookGroupIds}
              classes={classes}
            />
          ) : (
            <Typography variant="subtitle1">
              No hook groups are defined
            </Typography>
          ))}
        <Button
          spanProps={{ className: classes.actionButton }}
          tooltipProps={{ title: 'Create Hook' }}
          color="secondary"
          variant="round"
          onClick={this.handleCreateHook}>
          <PlusIcon />
        </Button>
      </Dashboard>
    );
  }
}
