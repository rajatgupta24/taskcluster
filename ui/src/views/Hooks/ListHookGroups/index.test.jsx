import React from 'react';
import { render, waitFor, act } from '@testing-library/react';
import { ApolloProvider } from 'react-apollo';
import setupClient from 'apollo-client-mock';
import { MemoryRouter } from 'react-router-dom';
import ListHookGroups from './index';

const typeDefs = `
  type User {
    id: ID!
  }
`;

it('should render ListHookGroupss page', async () => {
  const createClient = setupClient({}, typeDefs);

  await act(async () => {
    const { asFragment } = render(
      <MemoryRouter keyLength={0}>
        <ApolloProvider client={createClient()}>
          <ListHookGroups
            location={{
              search: {
                slice: jest.fn().mockReturnValue('search=test'),
              },
            }}
          />
        </ApolloProvider>
      </MemoryRouter>
    );

    await waitFor(() => {});
    expect(asFragment()).toMatchSnapshot();
  });
});
