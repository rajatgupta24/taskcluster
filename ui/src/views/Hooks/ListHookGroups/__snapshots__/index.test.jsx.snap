// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ListHookGroupss page 1`] = `<DocumentFragment />`;
